common-libs/common/src/services/DidDocumentService/ElemAnchoredDidDocument.ts

```diff
@@ -0,0 +1,28 @@
+import { KeyVault } from './KeyVault'
+import { DidResolver } from '../../shared/DidResolver'
+
+export default class ElemAnchoredDidDocument {
+  private readonly _keyVault: KeyVault
+  private readonly _signingKey: string
+
+  constructor(keyVault: KeyVault) {
+    this._signingKey = 'primary'
+    this._keyVault = keyVault
+  }
+
+  getMyDid(): string {
+    const { anchoredDid } = this._keyVault.metadata
+    return `${anchoredDid}`
+  }
+  getKeyId(did: string = null) {
+    return `${did ? did : this.getMyDid()}#${this._signingKey}`
+  }
+
+  async buildDidDocument(didResolver?: DidResolver) {
```

Comment:
Can we make this parameter non-optional in this PR?

Reply:
I'm not sure how to do it elegantly, as in case we make it non-optional it means it is required for each buildDidDocument even if it is not related to ElemAnchoredDidDocument
I don't have a good solution for it and finding out will take a time

Reply:
Yes, that would mean it is required for every call to `buildDidDocument`, because the caller cannot and should not know what kind of `DidDocumentService` it uses. That would be much more elegant than forcing callers to know when it is safe and when it is unsafe to call `buildDidDocument` without arguments.

It seems that, aside from possible tests, `buildDidDocument` is only used in an instance method of `Affinity` (so it can just pass `this._didResolver`) and in `BaseNetworkMember.register` (where we have all the data needed to create a `DidResolver` instance).